PROTO_tests/tests/pdarray_creation_test.py

```diff
@@ -0,0 +1,882 @@
+        t = ak.arange(bi - 1, bi + 9)
+        t_dup = ak.bigint_from_uint_arrays(t.bigint_to_uint_arrays())
+        assert t.to_list() == t_dup.to_list()
+        assert t_dup.max_bits == -1
+
+        # test setting max_bits after creation still mods
+        t_dup.max_bits = 200
+        assert t_dup.to_list() == [bi - 1, 0, 1, 2, 3, 4, 5, 6, 7, 8]
+
+        # test slice_bits along 64 bit boundaries matches return from bigint_to_uint_arrays
+        for i, uint_bits in enumerate(t.bigint_to_uint_arrays()):
+            slice_bits = t.slice_bits(64 * (4 - (i + 1)), 64 * (4 - i) - 1)
+            assert uint_bits.to_list() == slice_bits.to_list()
+
+    def test_arange(self):
+        assert [0, 1, 2, 3, 4] == ak.arange(0, 5, 1).to_list()
+        assert [5, 4, 3, 2, 1] == ak.arange(5, 0, -1).to_list()
+        assert [-5, -6, -7, -8, -9] == ak.arange(-5, -10, -1).to_list()
+        assert [0, 2, 4, 6, 8] == ak.arange(0, 10, 2).to_list()
+
+    def test_arange_dtype(self):
+        # test dtype works with optional start/stride
+        uint_stop = ak.arange(3, dtype=ak.uint64)
+        assert [0, 1, 2] == uint_stop.to_list()
+        assert ak.uint64 == uint_stop.dtype
+
+        uint_start_stop = ak.arange(3, 7, dtype=ak.uint64)
+        assert [3, 4, 5, 6] == uint_start_stop.to_list()
+        assert ak.uint64 == uint_start_stop.dtype
+
+        uint_start_stop_stride = ak.arange(3, 7, 2, dtype=ak.uint64)
+        assert [3, 5] == uint_start_stop_stride.to_list()
+        assert ak.uint64 == uint_start_stop_stride.dtype
+
+        # test uint64 handles negatives correctly
+        np_arange_uint = np.arange(2**64 - 5, 2**64 - 10, -1, dtype=np.uint64)
+        ak_arange_uint = ak.arange(-5, -10, -1, dtype=ak.uint64)
+        # np_arange_uint = array([18446744073709551611, 18446744073709551610, 18446744073709551609,
+        #        18446744073709551608, 18446744073709551607], dtype=uint64)
+        assert np_arange_uint.tolist() == ak_arange_uint.to_list()
+        assert ak.uint64 == ak_arange_uint.dtype
+
+        # test correct conversion to float64
+        np_arange_float = np.arange(-5, -10, -1, dtype=np.float64)
+        ak_arange_float = ak.arange(-5, -10, -1, dtype=ak.float64)
+        # array([-5., -6., -7., -8., -9.])
+        assert np_arange_float.tolist() == ak_arange_float.to_list()
+        assert ak.float64 == ak_arange_float.dtype
+
+        # test correct conversion to bool
+        expected_bool = [False, True, True, True, True]
+        ak_arange_bool = ak.arange(0, 10, 2, dtype=ak.bool)
+        assert expected_bool == ak_arange_bool.to_list()
+        assert ak.bool == ak_arange_bool.dtype
+
+        # test uint64 input works
+        uint_input = ak.arange(3, dtype=ak.uint64)
+        assert [0, 1, 2] == uint_input.to_list()
+        assert ak.uint64 == uint_input.dtype
+
+        # test int_scalars covers uint8, uint16, uint32
+        ak.arange(np.uint8(1), np.uint16(1000), np.uint32(1))
```

Comment:
@pierce314159 - should we add any validation here, or is this simply running evidence enough?